packages/admin-ui/pages/api/knex.ts

```diff
@@ -0,0 +1,30 @@
+interface KnexOptions {
+  dbType: string;
+  pgUrl?: string;
+  debug?: boolean;
+  sqlite_file?: string;
+}
+
+export const bootstrapKnex = (options: KnexOptions) => {
+  if (options.dbType === 'pg') {
+    const { pgUrl, debug } = options;
+    console.log('using pg');
+    /* eslint-disable @typescript-eslint/no-var-requires */
+    return require('knex')({
+      client: 'pg',
+      connection: pgUrl,
+      debug,
+      pool: { min: 2, max: 10 },
+    });
+  } else {
+    console.log('using sqlite');
+    /* eslint-disable @typescript-eslint/no-var-requires */
+    return require('knex')({
+      client: 'better-sqlite3',
+      connection: {
+        filename: options.sqlite_file,
+      },
+      useNullAsDefault: true,
+    });
+  }
+};
```

Comment:
Nextjs treats it as a route, the error only shows if we navigate to it, then it will call the page default function whether handling API or rendering page. It just builds the route initially.

If we command `nx build admin-ui`, we can see it will create a route page call `/api/knex`

<img width="538" alt="image" src="https://github.com/Canner/vulcan-sql/assets/9657305/9b802470-d86f-4fcc-a54c-220d3037995d">